src/main/utils/DocumentUtils.ts

```diff
@@ -2,29 +2,43 @@ import { AppRequest } from '../definitions/appRequest';
 import { EnglishOrWelsh, RespondentET3Model } from '../definitions/case';
 import { ApiDocumentTypeItem, DocumentTypeItem } from '../definitions/complexTypes/documentTypeItem';
 import { AllDocumentTypes, DefaultValues, languages } from '../definitions/constants';
+import { DocumentRow } from '../definitions/document';
 import { getLanguageParam } from '../helpers/RouterHelpers';
+import { dateInLocale } from '../helpers/dateInLocale';
 
 import CollectionUtils from './CollectionUtils';
+import DateUtils from './DateUtils';
 import NumberUtils from './NumberUtils';
 import ObjectUtils from './ObjectUtils';
 import RespondentUtils from './RespondentUtils';
 import StringUtils from './StringUtils';
 
 export default class DocumentUtils {
-  public static findET1DocumentByLanguage(
+  /**
+   * Tries to find ET1 form by the given language and document collection. First checks if document collection is empty
+   * or not. If empty returns undefined. If not empty tries to find ET1 form by given language. To discriminate ET1 form
+   * checks if document type is ET1 or not. If it finds any document with type ET1 according to selected language
+   * returns document. If selected language is Welsh and not able to find any ET1 form in Welsh but finds in English
+   * returns ET1 form in english.
+   * @param documentCollection is the all case documents that should have ET1 form.
+   * @param language is the parameter that we check for the document language.
+   * @return ET1 form by the given language in ApiDocumentTypeItemFormat.
+   */
+  public static findET1FormByLanguageAsApiDocumentTypeItem(
     documentCollection: ApiDocumentTypeItem[],
     language: string
   ): ApiDocumentTypeItem {
-    if (!documentCollection) {
+    if (CollectionUtils.isEmpty(documentCollection)) {
       return undefined;
     }
     for (const tmpDocument of documentCollection) {
       if (
```

Comment:
A bit hard to read... Is it possible to separate parts of the logic into some boolean functions?